tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs

```diff
@@ -54,27 +50,37 @@ private async Task<LabResponse> GetLabResponseFromApiAsync(UserQuery query)
                 throw new LabUserNotFoundException(query, "No lab user with specified parameters exists");
             }
 
-            return CreateLabResponseFromResultString(result);
+            return CreateLabResponseFromResultStringAsync(result).Result;
         }
 
-        private static LabResponse CreateLabResponseFromResultString(string result)
+        private async Task<LabResponse> CreateLabResponseFromResultStringAsync(string result)
         {
-            LabResponse[] responses = JsonConvert.DeserializeObject<LabResponse[]>(result);
-            if (responses.Length > 1)
+            LabUser[] userResponses = JsonConvert.DeserializeObject<LabUser[]>(result);
+            if (userResponses.Length > 1)
             {
                 throw new InvalidOperationException(
                     "Test Setup Error: Not expecting the lab to return multiple users for a query." +
                     " Please have rewrite the query so that it returns a single user.");
             }
 
-            var response = responses[0];
-            response.User.CredentialUrl = response.Lab.CredentialVaultkeyName;
-            response.User.TenantId = response.Lab.TenantId;
-            response.User.FederationProvider = response.Lab.FederationProvider;
+            var user = userResponses[0];
 
-            return response;
+            var appResponse = await GetLabResponseAsync(LabApiConstants.LabAppEndpoint + user.AppId).ConfigureAwait(false);
+            LabApp labApp = JsonConvert.DeserializeObject<LabApp>(StripBracketsFromLabResponse(appResponse));
```

Comment:
@bgavrilMS the new lab app endpoint also adds extra brackets like the user endpoint used to. Ill ask gladwin to remove